c7n/resources/s3.py

```diff
@@ -431,75 +490,172 @@ def get_permissions(cls):
+            return augment_keys
+        elif augment_config == 'none':
+            return []
+        elif isinstance(augment_config, list):
+            return augment_config
+
+        for f in self.manager.iter_filters():
+            fkey = None
+            if not isinstance(f, ValueFilter):
                 continue
-            else:
-                log.warning(
-                    "Bucket:%s unable to invoke method:%s error:%s ",
-                    b['Name'], m, e.response['Error']['Message'])
-                # For auth failures, we don't bail out, continue processing if we can.
-                # Note this can lead to missing data, but in general is cleaner than
-                # failing hard, due to the common use of locked down s3 bucket policies
-                # that may cause issues fetching information across a fleet of buckets.
-
-                # This does mean s3 policies depending on augments should check denied
-                # methods annotation, generally though lacking get access to an augment means
-                # they won't have write access either.
-
-                # For other error types we raise and bail policy execution.
-                if e.response['Error']['Code'] == 'AccessDenied':
-                    b.setdefault('c7n:DeniedMethods', []).append(m)
+
+            f = f.data
+            # type: value
+            if f.get('type', '') == 'value':
+                fkey = f.get('key')
+            # k: v dict
+            elif len(f) == 1:
+                fkey = list(f.keys())[0]
+            if fkey is None:
+                continue
+
+            # remove any jmespath expressions
+            fkey = fkey.split('.', 1)[0]
+
+            # tags have explicit handling in value filters.
+            if fkey.startswith('tag:'):
+                fkey = 'Tags'
+
+            # denied methods checks get all keys
+            if fkey.startswith('c7n:DeniedMethods'):
+                return augment_keys
+
+            if fkey in augment_keys:
+                detected_keys.append(fkey)
+
+        return detected_keys
+
+    def get_client(self, region):
+        if region in self.region_clients:
+            return self.region_clients[region]
+        with self.session_lock:
+            self.region_clients[region] = self.session.client('s3', region=region)
+            return self.region_clients[region]
+
+    def normalize_region(self, location):
+        # for historical reason we normalize EU to eu-west-1
+        b_location = location.get('LocationConstraint')
+        # Location == region for all cases but EU
+        # https://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketGETlocation.html
+        if b_location is None:
+            b_location = "us-east-1"
+        elif b_location == 'EU':
+            b_location = "eu-west-1"
+            location['LocationConstraint'] = 'eu-west-1'
+
+    def assemble(self, bucket):
+
+        client = self.get_client(self.default_region)
```

Comment:
I'm wondering if we should use the bucket region by default here. This is most relevant for metrics which is a separate filter, but there can be cases where the bucket region has canonical data and other region info might be a bit delayed.

Reply:
we don't know the bucket region, till we do the get bucket location api call